docs/spec/typeddict.rst

```diff
@@ -637,3 +637,193 @@ In particular allowing ``Annotated[]`` to be the outermost annotation
+
+* For each item in ``B``, ``A`` has the corresponding key, unless the item in ``B`` is read-only, not required, and of top value type (``ReadOnly[NotRequired[object]]``).
+* For each item in ``B``, if ``A`` has the corresponding key, the corresponding value type in ``A`` is consistent with the value type in ``B``.
+* For each non-read-only item in ``B``, its value type is consistent with the corresponding value type in ``A``.
+* For each required key in ``B``, the corresponding key is required in ``A``.
+* For each non-required key in ``B``, if the item is not read-only in ``B``, the corresponding key is not required in ``A``.
+
+Discussion:
+
+* All non-specified items in a TypedDict implicitly have value type ``ReadOnly[NotRequired[object]]``.
+
+* Read-only items behave covariantly, as they cannot be mutated. This is similar to container types such as ``Sequence``, and different from non-read-only items, which behave invariantly. Example::
+
+    class A(TypedDict):
+        x: ReadOnly[int | None]
+
+    class B(TypedDict):
+        x: int
+
+    def f(a: A) -> None:
+        print(a["x"] or 0)
+
+    b: B = {"x": 1}
+    f(b)  # Accepted by type checker
+
+* A TypedDict type ``A`` with no explicit key ``'x'`` is not consistent with a TypedDict type ``B`` with a non-required key ``'x'``, since at runtime the key ``'x'`` could be present and have an incompatible type (which may not be visible through ``A`` due to structural subtyping). The only exception to this rule is if the item in ``B`` is read-only, and the value type is of top type (``object``). For example::
+
+    class A(TypedDict):
+        x: int
+
+    class B(TypedDict):
+        x: int
+        y: ReadOnly[NotRequired[object]]
+
+    a: A = { "x": 1 }
+    b: B = a  # Accepted by type checker
+
+Update method
+-------------
+
+In addition to existing type checking rules, type checkers should error if a TypedDict with a read-only item is updated with another TypedDict that declares that key::
+
+    class A(TypedDict):
+        x: ReadOnly[int]
+        y: int
+
+    a1: A = { "x": 1, "y": 2 }
+    a2: A = { "x": 3, "y": 4 }
+    a1.update(a2)  # Type check error: "x" is read-only in A
+
+Unless the declared value is of bottom type (:data:`~typing.Never`)::
+
+    class B(TypedDict):
+        x: NotRequired[typing.Never]
+        y: ReadOnly[int]
+
+    def update_a(a: A, b: B) -> None:
+        a.update(b)  # Accepted by type checker: "x" cannot be set on b
+
+Note: Nothing will ever match the ``Never`` type, so an item annotated with it must be absent.
+
+Keyword argument typing
+-----------------------
+
+:pep:`692` introduced ``Unpack`` to annotate ``**kwargs`` with a ``TypedDict``. Marking one or more of the items of a ``TypedDict`` used in this way as read-only will have no effect on the type signature of the method. However, it *will* prevent the item from being modified in the body of the function::
```

Comment:
Should ideally link to https://typing.readthedocs.io/en/latest/spec/callables.html#unpack-for-keyword-arguments, and perhaps this section should be incorporated there instead.